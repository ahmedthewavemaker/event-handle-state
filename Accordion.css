.Acc{
    margin: 10px;
    color: blue;
   
}

.Accordion{
    margin: 20px;
    padding: 5px;
}