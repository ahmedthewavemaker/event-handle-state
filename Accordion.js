import React, {Component} from 'react';
import './Accordion.css';

export default class Accordion extends Component {
    static defaultProps = { sections: [] };
    state = { currentSectionIndex: null };

    handleButtonClick = (sectionIndex) => {
        console.log('clicked');
        this.setState({ currentSectionIndex: sectionIndex })
    }

    renderItem(section, index, currentSectionIndex) {
        return (
            <li className='Acc' key ={index}>
                 <button 
                 type='button'
             onClick={() => this.handleButtonClick(index)}>
                {section.title}
            </button>
            {(currentSectionIndex === index) && <p>{section.content}</p>}
            </li>
        )
        
        }
    
    render() {
        const {currentSectionIndex} = this.state
        const {sections} = this.props
        return (
            <ul className='Accordion'>
                {sections.map((section, index) => 
                this.renderItem(section, index, currentSectionIndex))}
            </ul>
        )
    }

}
