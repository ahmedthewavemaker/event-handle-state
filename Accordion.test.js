import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]



describe('Accordion component', () => {
    it('renders empty without any Accordion given without errors', () => {
       const wrapper = shallow(<Accordion />)
       expect(toJson(wrapper)).toMatchSnapshot()
    })


    it('renders no section as default', () => {
        const wrapper = shallow(<Accordion section={sections}/>)
        expect(toJson(wrapper)).toMatchInlineSnapshot()
    })


    it('opens any clicked section', () => {
        const wrapper = shallow(<Accordion section={sections}/>)
        wrapper.find('button').at(1).stimulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    } )


    it('opens only one section at a time', () => {
        const wrapper = shallow(<Accordion section={sections}/>)
        wrapper.find('button').at(1).stimulate('click')
        wrapper.find('button').at(2).stimulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })




})


