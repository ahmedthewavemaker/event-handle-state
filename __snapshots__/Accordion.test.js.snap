// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component renders empty without any Accordion given without errors 1`] = `
<ul
  className="Accordion"
/>
`;
